internal_workflow_testsuite.go

```diff
@@ -257,19 +258,34 @@ func (env *testWorkflowEnvironmentImpl) newTestWorkflowEnvironmentForChild(optio
 	return childEnv
 }
 
-func (env *testWorkflowEnvironmentImpl) setActivityTaskList(tasklist string, activityFns ...interface{}) {
+func (env *testWorkflowEnvironmentImpl) setActivityTaskList(
+	tasklist string,
+	ao RegisterActivityOptions,
+	activityFns ...interface{},
+) {
 	for _, activityFn := range activityFns {
-		env.testSuite.RegisterActivity(activityFn)
+		env.getHostEnv().RegisterActivityWithOptions(activityFn, ao)
```

Comment:
What use case does that solve? It's very beneficial to have self-contained tests. Tests should register workflows/activities that they are using. We can always use reusable helper methods to make that easier.